File: scripts/check_player_stats.js
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';

dotenv.config();

const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

async function checkPlayerStats() {
  try {
    // Fetch the 10 most recently updated players with their stats
    const { data: players, error } = await supabase
      .from('players')
      .select('name, sport, stats, updated_at')
      .not('stats', 'is', null)
      .order('updated_at', { ascending: false })
      .limit(10);

    if (error) {
      console.error('Error fetching players:', error);
      return;
    }

    if (!players || players.length === 0) {
      console.log('No players with stats found.');
      return;
    }

    console.log('\n=== Player Statistics ===\n');
    
    players.forEach((player, index) => {
      console.log(`\n${index + 1}. ${player.name} (${player.sport})`);
      console.log('Updated at:', player.updated_at);
      console.log('Stats:', JSON.stringify(player.stats, null, 2));
      console.log('---');
    });
    
    console.log('\n=== End of Results ===\n');

  } catch (err) {
    console.error('Unexpected error:', err);
  }
}

checkPlayerStats();

File: scripts/check_player_stats.py
import os
from supabase import create_client
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Initialize Supabase client
supabase = create_client(
    os.getenv("PUBLIC_SUPABASE_URL"),
    os.getenv("PUBLIC_SUPABASE_KEY")
)

def check_players():
    """Check if there are any players in the database."""
    try:
        # Get a sample of players
        response = supabase.table('nba_players').select('*').limit(5).execute()
        
        if hasattr(response, 'data') and response.data:
            print(f"Found {len(response.data)} players in the database.")
            print("Sample players:")
            for player in response.data:
                print(f"- {player.get('first_name')} {player.get('last_name')} (ID: {player.get('external_player_id')})")
            return response.data
        else:
            print("No players found in the database.")
            return []
            
    except Exception as e:
        print(f"Error checking players: {str(e)}")
        return []

def check_player_stats(player_id: str):
    """Check if a specific player has stats in the database."""
    try:
        response = supabase.table('nba_player_season_stats') \
            .select('*') \
            .eq('external_player_id', player_id) \
            .execute()
            
        if hasattr(response, 'data') and response.data:
            print(f"Found stats for player ID {player_id}:")
            print(response.data[0])
            return response.data[0]
        else:
            print(f"No stats found for player ID {player_id}")
            return None
            
    except Exception as e:
        print(f"Error checking player stats: {str(e)}")
        return None

if __name__ == "__main__":
    print("=== Checking NBA Players and Stats ===")
    players = check_players()
    
    if players:
        # Check stats for the first player
        first_player_id = players[0].get('external_player_id')
        if first_player_id:
            print("\nChecking stats for first player...")
            check_player_stats(first_player_id)
    
    print("\n=== Check Complete ===")
